### ISSUE TITLE
Next.js redirects in config broken in rc15

### ISSUE THREAD

**Issue by cmanou** [sst/open-next#404]
In 3.0.0rc15 redirects in next config such as :
```
    redirects: () => [{ source: '/settings', destination: '/settings/general', permanent: false }],
```
No longer seem to work they are working in rc14, my open-next config:
```
default: {
    override: {
      wrapper: 'aws-lambda-streaming',
    },
  },
 ```
From the lambda logs it looks like its being invoked 
<img width="1133" alt="Screenshot 2024-04-15 at 12 06 32 PM" src="https://github.com/sst/open-next/assets/683160/1b4fa60c-6854-4a1b-81bd-5575920fa003">
But the browser just hangs and says the connection is lost for safari and in chrome it takes 20s which is some timeout.

**Comment by conico974** [comment]
Could you create a minimum repro ? 
rc14 is the one with broken redirect for me, and rc15 does work for all the e2e tests with redirect
https://github.com/sst/open-next/blob/v3-rc/examples/app-router/next.config.js

**Comment by cmanou** [comment]
Okay reproduction here https://github.com/cmanou/ion-open-next-reproduction its currently deployed if you go here `https://d786ln7hnzo0y.cloudfront.net/settings` in safari it just does this
<img width="663" alt="Screenshot 2024-04-18 at 9 28 41 PM" src="https://github.com/sst/open-next/assets/683160/3a86b80d-b766-4c18-a456-42a12d93312d"> in chrome it waits for the function timeout and then redirects after 20 seconds
<img width="881" alt="Screenshot 2024-04-18 at 9 29 18 PM" src="https://github.com/sst/open-next/assets/683160/4cb47413-614e-4197-9834-d39b1f6d34f6">

**Comment by conico974** [comment]
Link to the discord threads discussing this https://discord.com/channels/983865673656705025/1230482660913184800

TLDR; This is an issue with lambda itself and depending on the region and the account, you might not experience this issue at all. Lambda is also probably rolling out the fix

**Comment by conico974** [comment]
Fixed in [3.0.3](https://github.com/sst/open-next/releases/tag/v3.0.3)
Be aware that there is another issue with streaming that cannot be fixed by us but only by aws. See https://discord.com/channels/983865673656705025/1249368592558985247
